backend-rust/src/graphql_api/node_status.rs

```diff
@@ -0,0 +1,497 @@
+use super::{get_config, get_pool, ApiError, ApiResult};
+use async_graphql::{connection, types, ComplexObject, Context, Enum, Object, SimpleObject};
+use reqwest::{Client, StatusCode};
+use serde::{Deserialize, Serialize};
+use std::{
+    cmp::{min, Ordering::Equal},
+    time::Duration,
+};
+use tokio::sync::watch::{Receiver, Sender};
+use tokio_util::sync::CancellationToken;
+use tracing::{error, info};
+
+#[derive(Default)]
+pub(crate) struct QueryNodeStatus;
+
+#[allow(clippy::too_many_arguments)]
+#[Object]
+impl QueryNodeStatus {
+    async fn node_statuses(
+        &self,
+        ctx: &Context<'_>,
+        sort_direction: NodeSortDirection,
+        sort_field: NodeSortField,
+        #[graphql(desc = "Returns the first _n_ elements from the list.")] first: Option<u64>,
+        #[graphql(desc = "Returns the elements in the list that come after the specified cursor.")]
+        after: Option<String>,
+        #[graphql(desc = "Returns the last _n_ elements from the list.")] last: Option<u64>,
+        #[graphql(desc = "Returns the elements in the list that come before the specified cursor.")]
+        before: Option<String>,
+    ) -> ApiResult<connection::Connection<String, NodeStatus>> {
+        let config = get_config(ctx)?;
+        let pool = get_pool(ctx)?;
+        let handler =
+            ctx.data::<Receiver<Option<Vec<NodeStatus>>>>().map_err(ApiError::NoReceiver)?;
+        if first.is_some() && last.is_some() {
+            return Err(ApiError::QueryConnectionFirstLast);
+        }
+
+        let mut statuses = if let Some(statuses) = handler.borrow().clone() {
+            statuses
+        } else {
+            Err(ApiError::InternalError("Node collector backend has not responded".to_string()))?
+        };
+
+        statuses.sort_by(|a, b| {
+            let ordering = match sort_field {
+                NodeSortField::AveragePing => a.average_ping.partial_cmp(&b.average_ping),
+                NodeSortField::BlocksReceivedCount => {
+                    a.blocks_received_count.partial_cmp(&b.blocks_received_count)
+                }
+                NodeSortField::ClientVersion => a.client.partial_cmp(&b.client),
+                NodeSortField::ConsensusBakerId => {
+                    a.consensus_baker_id.partial_cmp(&b.consensus_baker_id)
+                }
+                NodeSortField::FinalizedBlockHeight => {
+                    a.finalized_block_height.partial_cmp(&b.finalized_block_height)
+                }
+                NodeSortField::NodeName => a.node_name.partial_cmp(&b.node_name),
+                NodeSortField::PeersCount => a.peers_count.partial_cmp(&b.peers_count),
+                NodeSortField::Uptime => a.uptime.partial_cmp(&b.uptime),
```

Comment:
Really minor thing here, and it probably does not matter so feel free to ignore, but we could use `cmp` and [`f64::total_cmp`](https://doc.rust-lang.org/std/primitive.f64.html#method.total_cmp) to avoid the `unwrap_or` later